File: Java/Count the Characters V3.java
//Count the Characters v3

import java.util.Scanner;

public class Characters{
  public int countCharacters(String input, char characterInput){
    int count = 0;
    int length = input.length();
    for(int counter = 0; counter < length; counter++){
      char container= input.charAt(counter);
      char smallContainer = Character.toLowerCase(container);
      char smallCharacter = Character.toLowerCase(characterInput);
      if(smallContainer == smallCharacter){
        count++;
      }
    }
    return count;
  }
  public static void main(String[] args){
    Scanner scanner = new Scanner(System.in);
    Characters characters = new Characters();
    System.out.println("Count the Characters v3");
    System.out.print("Enter a string:");
    String word = scanner.nextLine();
    if(word.equals("")){
      System.out.println("No input, please input a string.");
    } else{
      System.out.print("Enter a character to count in [" + word + "]:");
      String letter = scanner.nextLine();
      if(letter.isEmpty()){
        System.out.println("No input, please input a character to count.");
      } else{
        char finalLetter = letter.charAt(0);
        int result = characters.countCharacters(word, finalLetter);
        System.out.println(letter + " occurs " + result + " times in " + word + ".");
      }
    }
  }
}

File: C++/Count_the_Characters_V3.cpp
#include <iostream>
#include <string>
#include <cctype>
using namespace std;

int countCharacters(string input, char characterInput) {
    int count = 0;
    int length = input.length();

    for (int counter = 0; counter < length; counter++) {
        char container = input[counter];
        char smallContainer = tolower(container);
        char smallCharacter = tolower(characterInput);
        if (smallContainer == smallCharacter) {
            count++;
        }
    }

    return count;
}

int main() {
    string word;
    cout << "Count the Characters v3" << endl;
    cout << "Enter a string:";
    getline(cin, word);

    if (word == "") {
        cout << "No input, please input a string." << endl;
    } else {
        cout << "Enter a character to count in [" << word << "]:";
        string letter;
        getline(cin, letter);

        if (letter.empty()) {
            cout << "No input, please input a character to count." << endl;
        } else {
            char finalLetter = letter[0];
            int result = countCharacters(word, finalLetter);
            cout << letter << " occurs " << result << " times in " << word << "." << endl;
        }
    }

    return 0;
}
